### PR TITLE
WIP: Add dark color scheme

### PR DESCRIPTION
**Pull request by tysongach**

### CODE REVIEW

File: src/settings/lib/color.scss [hunk 1]
```diff
@@ -3,3 +3,23 @@
...
+  --tbds-color-viewport-background: #fafafa;
+  --tbds-color-text-default: #{$tbds-brand-gray};
+  --tbds-color-text-secondary: #{$tbds-gray-900};
+  --tbds-color-text-link: #{$tbds-brand-gray};
+  --tbds-color-text-link-hover: #{$tbds-brand-red};
+  --tbds-color-border-default: rgba(150, 150, 150, 0.2);
+  --tbds-color-input-background: rgba(150, 150, 150, 0.2);
+}
+
+@media (prefers-color-scheme: dark) {
```
[the remark this answers is not in the document]
Comment (tysongach) [reply src/settings/lib/color.scss]: I did some digging and this appears to be [a bug](https://github.com/stylelint/stylelint/pull/3951) that was fixed in stylelint v10, but [Hound is running v9.10.1](http://help.houndci.com/en/articles/2461415-supported-linters). I messaged the Hound team to see if they can add/upgrade to v10.

**Comment by tysongach** [comment]
Spoke with @sneamonitakis IRL about this…

It was fun to experiment with this, but we probably want to hold off for a bit on merging the dark mode specific changes, because it’s a bit of an “all or nothing” situation. If we merged that into Hub, the app would visually break in odd ways, with some components changing their appearance, while all of the custom CSS components in the app, not.

That said, this PR includes some move of variables over from Sass and into CSS Custom Properties, which is good and fine to do any time. So we can extract some of that good stuff into separate PRs merge-able today, and save the dark mode specific stuff for later.

File: src/settings/lib/color.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (tysongach) [reply src/settings/lib/color.scss]: The Hound team has upgraded to stylelint v10.x.

**Comment by tysongach** [comment]
Thinking out loud:

- How might we allow consumers (applications using the system) _opt-in_ to dark-mode?
- It may be wise to also add the `color-scheme` property, which signals to the browser that we support both `light` and `dark` color schemes:

    ```css
    html {
      color-scheme: light dark;
    }
    ```

**Comment by ericwbailey** [comment]
I really like Andy Bell's approach to [dark mode and user choice](https://hankchizljaw.com/wrote/create-a-user-controlled-dark-or-light-mode/). Notably:

- Visual UI that informs the user it exists
- A toggle that allows either theme, which lets the user self-select their preference
- `localStorage` support for remembering that preference
- Scoped CSS Custom Properties

**Pull request closed without merging**